doc/MIGRATION_GUIDE.md

```diff
@@ -13,29 +13,15 @@ However because of initaliziation order (child logger has to be created after rc
 #include <moveit/utils/logger.hpp>
 ```
 
-Wherever the node is created there is the option to set the global logger for moveit using this syntax:
+Wherever the node is created there is the option to set the root logging namespace to be from the node for moveit using this syntax:
 
 ```C++
-moveit::setLogger(node->get_logger());
+moveit::setNodeLoggerName(node->get_name());
 ```
 
 Then wherever you call a logging macro you can use the `moveit::getLogger()` function:
 ```C++
-RCLCPP_INFO(moveit::getLogger(), "Very important info message");
-```
-
-To have namespaces you need to create a child logger.
-There is a function for that too.
-This creates a child of the global logger.
-You'll find this in the constructor of many of our classes.
-
-```C++
-, logger_(moveit::makeChildLogger("servo"))
-```
-
-Once you have a child logger you can use it in logging macros:
-```C++
-RCLCPP_INFO(logger_, "Very important info message");
+RCLCPP_INFO(moveit::getLogger("my_namespace"), "Very important info message");
```

Comment:
Honestly very stupid question:
```suggestion
moveit::logInfo("my_namespace", "Very important info message");
```
or
```suggestion
MOVEIT_INFO("my_namespace", "Very important info message");
```
do read loads better, right?

I would clearly support a readable wrapper over an awkward combination of MoveIt internal static magic (definitely not what god intended) TOGETHER with C macro magic (which was clearly invented by demons).

Reply:
Here is the issues I see with those two approaches.

## functions

The macros capture the file, function, and line number from the place where the macro is called. If we wrapped the ros logging system in our own functions we would loose that.

## our own macros

This is a more ergonomic solution but would require me to write macros that wrap the ros macros. That would be a lot of code to write, test, and maintain. I think this approach is better given that tradeoff.

Reply:
Fair enough. We could probably limit a convenience macro wrapper to the plain variants for almost all uses. [That approach works well in OMPL](https://ompl.kavrakilab.org/group__logging.html#ga576d0bc79b521f19c5415f330e2b173d) and surely others. But then the question would be whether stream or printf-style and combinations might quickly blow up again if people disagree.

Still, an argument slot that only ever contains one generic term in correct usage should not exist either.

Between a rock and a hard place by design.

Reply:
Yea, I think in this case the ROS 2 logging interface is just worse from a usability standpoint. I want to reduce the amount of abstractions I maintain so I don't want to maintain a set of wrapper logging macros. For that reason I think what is in this PR is the best compromise for now.